bittensor/_axon/__init__.py

```diff
@@ -427,34 +432,65 @@ def check_signature(
             previous_nonce = self.nonces[endpoint_key]
             # Nonces must be strictly monotonic over time.
             if nonce <= previous_nonce:
-                raise Exception("Nonce is too small")
+                raise GrpcException("Nonce is too small")
 
         if not keypair.verify(message, signature):
-            raise Exception("Signature mismatch")
+            raise GrpcException("Signature mismatch")
         self.nonces[endpoint_key] = nonce
 
+    def request_deserialization(
+            self,
+            request
+    ):
+        r"""deserialization of request in payload. Used for authenication"""
+        
+        try: 
+            synapse = bittensor.synapse.deserialize(request.synapses[0])
+            request_matrix = synapse.deserialize_forward_request_tensor(request.tensors[0])
```

Comment:
Shouldn't take this into consideration if this is a `Forward` or a `Backward` ? You have that in the `method_name` at the call site.